### DISCUSSION TITLE
How PB authenticate user? How cookie pb_auth look like

### DISCUSSION THREAD

**Discussion by phulelouch** [pocketbase/pocketbase#1182] [Q&A]
Hi, 
I am doing a project where I don't wanna use pb_JS_SDK,
I understand when set rule @request.auth.id != "" then only register user can access that API
But how do you guys do authentication? Cookie or Authentication header?
If Cookie then how the pb_auth cookie look like so I can implement them
Thank you

**Comment by ganigeorgiev** [comment]
We are not using session cookies. PocketBase uses JWT tokens for the authentication that need to be set in the `Authorization` header with each request (aka. `Authorization: TOKEN`).

For more information you can check https://pocketbase.io/docs/authentication/.

To generate an admin token, you can check the "Admins > Auth with password" api - https://pocketbase.io/docs/api-admins/.

To generate an app user token, you can check the "Records > Auth with passowrd/Auth with OAuth2" apis - https://pocketbase.io/docs/api-records/#auth-record-actions

**Comment by adharshmk96** [comment]
Usually when I do web based applications, I store tokens in http only cookie thinking of security. 

Maybe some browser extension could potentially steal users's token right ? Would it be a good idea to add a http only cookie support ?

I can access the token by calling 
```
localStorage.getItem("__pb_superuser_auth__")
```

**Reply by ganigeorgiev** [reply]
Cookies comes with their own set of problems and I don't plan introducing them. If users think they need cookies, then you can implement it on your own by registering a custom middleware via Go or the JSVM.

Regarding the local storage security, please check https://github.com/pocketbase/js-sdk?tab=readme-ov-file#security (_the Dashboard UI is served with a basic CSP header_).
